experiments/ssl4eo/download_ssl4eo.py

```diff
@@ -163,78 +161,63 @@ def adjust_coords(
     ]
 
 
-def get_properties(image: ee.Image) -> Any:
-    return image.getInfo()
-
-
 def get_patch(
     collection: ee.ImageCollection,
     center_coord: list[float],
     radius: float,
     bands: list[str],
-    crop: Optional[dict[str, Any]] = None,
+    original_resolutions: list[int],
+    new_resolutions: list[int],
     dtype: str = "float32",
 ) -> dict[str, Any]:
-    image = collection.sort("system:time_start", False).first()  # get most recent
-    region = (
-        ee.Geometry.Point(center_coord).buffer(radius).bounds()
-    )  # sample region bound
-    reproj_bands = "B8"
-    reproj_resolution = 30
-    other_bands = [b for b in bands if b != reproj_bands]
-    patch_reproj = (
-        image.select(reproj_bands)
-        .reproject(
-            crs=image.select(reproj_bands).projection().crs(), scale=reproj_resolution
-        )
-        .sampleRectangle(region, defaultValue=0)
-    )
-    patch_other = image.select(*other_bands).sampleRectangle(region, defaultValue=0)
-    # patch = image.select(*bands).sampleRectangle(region, defaultValue=0)
-    # features = patch.getInfo()  # the actual download
-    features_reproj = patch_reproj.getInfo()
-    features_other = patch_other.getInfo()
-
-    raster = OrderedDict()
-    for band in bands:
-        if band == reproj_bands:
-            img = np.atleast_3d(features_reproj["properties"][band])
-        else:
-            img = np.atleast_3d(features_other["properties"][band])
-        if crop is not None:
-            img = center_crop(img, out_size=crop[band])
-        raster[band] = img.astype(dtype)
-
-    coords0 = np.array(features_other["geometry"]["coordinates"][0])
+    image = collection.sort("CLOUD_COVER").first()
+    region = ee.Geometry.Point(center_coord).buffer(radius).bounds()
+
+    # Group by original and new resolution
+    band_groups = defaultdict(list)
+    for i in range(len(bands)):
+        band_groups[(original_resolutions[i], new_resolutions[i])].append((i, bands[i]))
+
+    # Reproject (if necessary) and download all bands
+    raster = {}
+    for (orig_res, new_res), value in band_groups.items():
+        indices, bands = zip(*value)
+        patch = image.select(*bands)
+        if orig_res != new_res:
+            patch = patch.reproject(patch.projection().crs(), scale=new_res)
+        patch = patch.sampleRectangle(region, defaultValue=0)
+        features = patch.getInfo()
+        for i, band in zip(indices, bands):
+            x = features["properties"][band]
+            x = np.atleast_3d(x)
+            x = center_crop(x, out_size=2 * radius // new_res)
```

Comment:
Instead of the user providing the crop size, we compute the appropriate crop size based on the resolution of each band.